### PR TITLE
feat: [CDS-49721]: Add digest field for DockerRegistry artifacts

### PR DESCRIPTION
**Pull request by vcalasansh**
## Describe your changes
This PR adds a new optional parameter called `digest` for docker registry artifacts in Harness NG. This feature is behind the Feature Flag called `SPG_NG_DOCKER_ARTIFACT_DIGEST`.

If this parameter is populated by the user, its value will be available in the variable `<+artifact.digest>` and during the "Service" step of the deployment stage, we will validate the selected digest against available ones in the docker repository. 

If the selected digest doesn't match, we will show a validation error during the "Service" step, explaining that the selected SHA256 digest for this image does not match the possible ones fetched from docker registry.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42238)

### CODE REVIEW

**Review by sribalij — APPROVED** [review]
> Reviewed 30 of 30 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42238)* status: 30 file(s) to review. 1 of 2 approvals obtained.

**Review by sribalij — APPROVED** [review]
> Reviewed 6 of 6 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42238)* status: Some of the checks are too old. 30 file(s) to review. 1 of 2 approvals obtained.

**Review by puthrayaharness — APPROVED** [review]

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 23 of 30 files at r1, 6 of 6 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42238)* status: Some of the checks are too old.

**Comment by vcalasansh** [comment]
trigger ft1,ft2

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/docker/DockerArtifactTaskHandler.java [hunk 1]
```diff
@@ -100,7 +100,7 @@ public ArtifactTaskExecutionResponse validateArtifactImage(DockerArtifactDelegat
   public ArtifactMetaInfo getArtifactMedataInfo(DockerArtifactDelegateRequest attributesRequest) {
     return dockerRegistryService.getArtifactMetaInfo(
         DockerRequestResponseMapper.toDockerInternalConfig(attributesRequest), attributesRequest.getImagePath(),
-        attributesRequest.getTag());
+        attributesRequest.getTag(), attributesRequest.getShouldFetchDockerV2DigestSHA256());
```
Comment (vcalasansh) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/docker/DockerArtifactTaskHandler.java:103]: This

**Pull request merged**